### PR TITLE
encryptionkeyrotation: Add volume capabilities

### PR DESCRIPTION
**Pull request by black-dragon74**
This patch adds the option to pass the CSI
volume capabilities with the
EncryptionKeyRotateRequest.

### CODE REVIEW

File: encryptionkeyrotation/README.md [hunk 1]
```diff
@@ -77,11 +77,16 @@ message EncryptionKeyRotateRequest {
...
   string encryption_key = 3 [(csi.v1.csi_secret) = true];
 
+  // Volume capability describing how the CO intends to use this volume.
+  // This allows SP to determine if volume is being used as a block
+  // device or mounted file system. This is OPTIONAL.
+  csi.v1.VolumeCapability volume_capability = 4;
+
   // Plugin specific parameters passed in as opaque key-value pairs.
-  map<string, string> parameters = 4;
+  map<string, string> parameters = 5;
```
Comment (nixpanic) [inline encryptionkeyrotation/README.md:86]: you may not change the numbering of members in the message, you can only add new members with higher numbers

File: encryptionkeyrotation/README.md [hunk 2]
```diff
@@ -77,11 +77,16 @@ message EncryptionKeyRotateRequest {
   // This field is OPTIONAL
   string encryption_key = 3 [(csi.v1.csi_secret) = true];
 
+  // Volume capability describing how the CO intends to use this volume.
+  // This allows SP to determine if volume is being used as a block
+  // device or mounted file system. This is OPTIONAL.
+  csi.v1.VolumeCapability volume_capability = 4;
```
Comment (nixpanic) [inline encryptionkeyrotation/README.md:83]: Are the important attributes for this not available through the `volume_id` already? Or is there not an other way to identify/detect this while performing the `EncryptionKeyRotateRequest`?

File: encryptionkeyrotation/README.md (hunk 1 above)
Comment (black-dragon74) [reply encryptionkeyrotation/README.md:86]: I am aware of this, I did this as the implementation is not live yet. Shall I revert?

File: encryptionkeyrotation/README.md (hunk 2 above)
Comment (black-dragon74) [reply encryptionkeyrotation/README.md]: Marking it as resolved wrt offline discussions.

**Review by Madhu-1 — APPROVED** [review]

**Review by nixpanic — APPROVED** [review]

**Pull request merged**